app/controllers/results_submission_controller.rb

```diff
@@ -97,6 +97,75 @@ def upload_json
     end
   end
 
+  def import_from_live
+    @competition = competition_from_params
+    return redirect_to competition_submit_results_edit_path if @competition.results_submitted?
+
+    errors = []
+    results_to_import = @competition.rounds.flat_map do |round|
+      round.round_results.map do |result|
+        results_to_import >> InboxResult.new({
```

Comment:
You shouldn't be appending anything here anymore. Just `InboxResult.new` directly as return value will do.